tests/resolver.rs

```diff
@@ -0,0 +1,121 @@
+//! Test that syn-inline-mod can resolve this crate's lib.rs properly.
+
+use std::path::{Path, PathBuf};
+use syn::Item;
+use syn_inline_mod::{find_mod_path, InlinerBuilder};
+
+#[test]
+fn resolve_lib() {
+    let builder = InlinerBuilder::new();
+
+    let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+    let lib_rs = manifest_dir.join("src/lib.rs");
+
+    let (_, files_seen) = inline(&builder, &lib_rs);
+
+    // Ensure that the list of files is correct.
+    let file_list: Vec<_> = files_seen
+        .iter()
+        .map(|(path, _)| {
+            let rel_path = path
+                .strip_prefix(manifest_dir)
+                .expect("path should be relative to manifest dir");
+            rel_path.to_str().expect("path is valid Unicode")
+        })
+        .collect();
+
+    // The order visited should be the same as the order in which "mod" statements are listed.
```

Comment:
This is a sane/reasonable expectation, but is it a requirement, or is the requirement just that all the files be visited?

Reply:
It's not a requirement for my specific use case, but I think it's reasonable to document this as part of the API -- what do you think?

Reply:
Oh never mind, this falls out naturally from the fact that we don't change the order of modules, heh.